examples/pageon-metric.js

```diff
@@ -0,0 +1,33 @@
+import { browser } from 'k6/x/browser/async';
+
+export const options = {
+  scenarios: {
+    ui: {
+      executor: 'shared-iterations',
+      options: {
+        browser: {
+            type: 'chromium',
+        },
+      },
+    },
+  },
+}
+
+export default async function() {
+  const page = await browser.newPage();
+
+  page.on('metric', (msg) => {
+    msg.groupURLTag({
+      groups: [
+        {url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, name:'test'},
+      ]
+    });
+  });
```

Comment:
I'm not sure of the naming of the API (method, fields, etc). Any suggestions?

Reply:
I'd suggest:

```
  page.on('metric', m => {
    m.tagURLs({
      urls: [
        {url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, name:'test'},
      ]
    });
  });
```

Or, even:

```
  page.on('metric', m => {
    m.tagURLs([
      { url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, name:'test' },
    ]);
  });
```

Reply:
+1 for `urls` instead of `groups` 👍

Reply:
What about `groupTags` instead of `groupURLTag`?:

  page.on('metric', (msg) => {
    msg.groupTags({
      urls: [
        {url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, name:'test'},
      ]
    });
  });

We're not duplicating the word url as much, and we're allowing for it to be extended in case we need to group another tag in the future.

Reply:
Are we grouping tags, or are we telling k6 to tag metrics that match the criteria(s) like URLs? So, our goal here is not to "group tags"; instead, "tagging the metrics" (not even URLs in general) so that k6 can automatically group them. That's why I suggested `tagURLs`. Now I think calling it `tag` might be better after seeing your reasoning 👍 

Something like:

```javascript
  page.on('metric', metric => {
    // give a tag to the metric that matches the URL
    metric.tag({
      urls: [
        {url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, name:'test'},
      ]
    });
  });
```

_PS: Making it extendable is good, but it also complicates the API :)_

Reply:
> JavaScript is not like Go. We can detect whether users passing an Object or an array without making any breaking changes. Making the API extendable is good, but it also complicates it. So, I'd vote for keeping the API simpler rather than complicating it for unforeseen future needs.

That's true, but I think that's messy and not a great DX. It complicates other things too, such as the implementation in go, type defs and documentation. I'd still prefer to keep the object rather than an array.

> By definition, tagging means grouping in k6 terms.

Yeah I think that makes sense 👍 

> Saying "group" is also a bit confusing as k6 has a group concept.

I don't quite align with you here, but since i agree with what you mean by tagging, i'm happy to drop `group`.

> Are we grouping tags, or are we telling k6 to tag metrics that match the criteria(s), like URLs?

Good bit of analysis on what we're trying to do here, and I agree.

I like the latest suggestion 👍 

```js
  page.on('metric', metric => {
    // give a tag to the metrics that matches the URL
    metric.tag({
      urls: [
        {url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, name:'test'},
      ]
    });
  });
```

> // getting a `msg` on a `metric` sounds a bit weird?

It's technically an event, so i think msg is fine, we're not passing the `metric` itself to the event handler. Happy with metric though as it better represents where the action will take place and easier to reason with 👍

Reply:
Changed to the latest suggestion in 597ddd7e65d2edf694206edd2431639ed27df5d4

Reply:
Changes `msg` to `metric` in 72d59070025cb4a645d9db117a4a9d532aca755b

Reply:
Thanks :)